src/store/login/login.thunks.ts

```diff
@@ -0,0 +1,305 @@
+import { keccak256 } from "js-sha3";
+import hermez from "@hermeznetwork/hermezjs";
+import { push } from "connected-react-router";
+import { utils } from "ethers";
+import WalletConnectProvider from "@walletconnect/web3-provider";
+import { Web3Provider } from "@ethersproject/providers";
+import { Signer } from "@ethersproject/abstract-signer";
+import { isEnvironmentSupported } from "@hermeznetwork/hermezjs/src/environment";
+import * as globalActions from "src/store/global/global.actions";
+import * as globalThunks from "src/store/global/global.thunks";
+import * as loginActions from "src/store/login/login.actions";
+import { TREZOR_MANIFEST_MAIL } from "src/constants";
+import { buildEthereumBIP44Path } from "src/utils/hw-wallets";
+import { getNextForgerUrls } from "src/utils/coordinator";
+import { RootState } from "src/store";
+import { AppDispatch, AppThunk } from "src";
+// domain
+import { SignerData, HermezWallet } from "src/domain/hermez";
+// persistence
+import * as persistence from "src/persistence";
+import { getAuthSignatures, setAuthSignatures } from "src/persistence/local-storage";
+
+async function getSignerData(
+  provider: Web3Provider,
+  walletName: loginActions.WalletName,
+  accountData: loginActions.AccountData
+): Promise<SignerData> {
+  switch (walletName) {
+    case loginActions.WalletName.METAMASK:
+    case loginActions.WalletName.WALLET_CONNECT: {
+      return {
+        type: "JSON-RPC",
+      };
+    }
+    case loginActions.WalletName.LEDGER: {
+      const chainId = (await provider.getNetwork()).chainId;
+      const { accountType, accountIndex } = accountData;
+      const path = buildEthereumBIP44Path(chainId, accountType, accountIndex);
+
+      return {
+        type: hermez.Signers.SignerType.LEDGER,
+        path,
+      };
+    }
+    case loginActions.WalletName.TREZOR: {
+      const chainId = (await provider.getNetwork()).chainId;
+      const { accountType, accountIndex } = accountData;
+      const path = buildEthereumBIP44Path(chainId, accountType, accountIndex);
+
+      return {
+        type: hermez.Signers.SignerType.TREZOR,
+        path,
+        manifest: {
+          email: TREZOR_MANIFEST_MAIL,
+          appUrl: window.location.origin,
+        },
+      };
+    }
+    default: {
+      throw new Error("Wallet not supported");
+    }
+  }
+}
+
+/**
+ * Helper function that signs the authentication message depending on Wallet type
+ * @param {Web3Provider | Signer} providerOrSigner
+ * @param {String} message
+ * @param {String} address
+ * @returns {Promise<unknown>} A promise that resolves to the signature
```

Comment:
Isn't this Promise<string>? Also, with TS, do we need params and returns docs?

Reply:
This should be `Promise<string>`.

> Also, with TS, do we need params and returns docs?

I'm glad you mention it! No!! JSDocs are completely useless now and even counterproductive since they use to get obsoleted and lie to us.

Can I remove them as I go?

Reply:
I would remove only params and returns but keep function descriptions. But yeah, go ahead :)

Reply:
Cool!